requirements-common.txt

```
@@ -12,6 +12,6 @@ pydantic >= 2.0  # Required for OpenAI server.
 prometheus_client >= 0.18.0
 tiktoken == 0.6.0  # Required for DBRX tokenizer
 lm-format-enforcer == 0.9.3
-outlines == 0.0.34 # Requires torch >= 2.1.0
+outlines == 0.0.37 # Requires torch >= 2.1.0
```

I'm fine doing a `>=` pin to support future versions. (outlines rapid release schedules!) 

```suggestion
outlines >= 0.0.37
```

---

I’m a big fan of pinning to one specific version to make sure this code will still work in a few months. Then one can work with dependabot or similar to keep up with changes without too much manual effort. 

But I don’t want to force this. 

Maybe there should be a project-wide policy in this regard?

let me know what I should do here.

---

we don't have a project wide strict pin policy. i prefer string pin is most cases. the main concern is now outlines also depend on vllm versions so i want to avoid as much circular dependency as possible by relaxing it (see https://github.com/vllm-project/vllm/issues/4153)

---

That's a very good reason for unpinning.